Fix typescript interface

jonmagic opened the pull request:
Fixes `Type 'ApolloClient<NormalizedCache>' is not assignable to type 'ApolloClient<Cache>'.` errors.

My typescript foo is weak so I'm not sure if I'm doing what @corydeppen suggested in https://github.com/apollographql/react-apollo/issues/1299#issuecomment-340275539? My changes worked for me at least.

Closes https://github.com/apollographql/react-apollo/issues/1299

estaub commented:
In case it wasn't clear from the spewage above, if you drill down into the Travis log the problem is simply 
`No CHANGELOG added.`

It'll probably be hard to get folks to look at it till you fix this nit.

jonmagic commented:
@corydeppen my typescript foo is weak, are the changes here what you suggested? I haven't taken the time to figure out the `TCache` suggestion, just fixed the error I was getting and moved on.

I could also use some help with a description for [the Changelog](https://github.com/jonmagic/react-apollo/blob/cff620b2e057d5baa5df956274a5537cf028b5a6/Changelog.md#change-log). The best I've come up with so far is:

> Fixes ApolloClient cache type bug

estaub commented:
Not sure, you may want to bail; see https://github.com/apollographql/react-apollo/pull/1319

jonmagic commented:
Closing in favor of https://github.com/apollographql/react-apollo/pull/1319
